components/Starknet/modules/architecture-and-concepts/pages/network-architecture/messaging-mechanism.adoc

```diff
@@ -222,7 +223,7 @@ Where:
 
 - `l1_handler` is a constant prefix, encoded in bytes (ASCII), as big-endian.
 - `chain_id` is a constant value that specifies the network to which this transaction is sent.
-- _h_ is the xref:architecture-and-concepts:cryptography/hash-functions.adoc#pedersen_hash[Pedersen] hash
+- _h_ is the xref:architecture-and-concepts:cryptography/hash-functions.adoc#pedersen_hash[Pedersen] hash (note that since we're hashing an array, the # of inputs needs to be appended to the hash).
```

Comment:
```suggestion
- _h_ is the xref:architecture-and-concepts:cryptography/hash-functions.adoc#pedersen_hash[Pedersen] hash (note that since we are hashing an array, the number of inputs needs to be appended to the hash).
```

Reply:
If it's a serialized array, `appended` may be misleading. The length of the array must be the first element of the array to be hashed.
`[len, c1, c2, c3, ...]`.

Reply:
@glihm Good catch. How about this:

```suggestion
- _h_ is the xref:architecture-and-concepts:cryptography/hash-functions.adoc#pedersen_hash[Pedersen] hash. Be aware that because an array is hashed, the length of the array must be the first element on the array.
```